### DISCUSSION TITLE
calibration of a winter saturated well with apparently strong evaporation effect in summer (France)

### DISCUSSION THREAD

**Discussion by VeloVolant** [pastas/pastas#837] [Q&A]
Hi ! I am facing difficulties to calibrate a head time serie from a well located in Auvergne ([link](https://ades.eaufrance.fr/Fiche/PtEau?Code=BSS001WHRD#mesures_graphiques)), France, within the volcanic range (about 1000 masl). The shaft well [is there/ggle street view](https://www.google.com/maps/@45.2420076,3.0988722,3a,64.8y,355.23h,77.59t/data=!3m7!1e1!3m5!1sSzyJLoy-MnjLlhKeWBfWXA!2e0!6shttps:%2F%2Fstreetviewpixels-pa.googleapis.com%2Fv1%2Fthumbnail%3Fpanoid%3DSzyJLoy-MnjLlhKeWBfWXA%26cb_client%3Dmaps_sv.tactile.gps%26w%3D203%26h%3D100%26yaw%3D71.2626%26pitch%3D0%26thumbfov%3D100!7i16384!8i8192?coh=205409&entry=ttu&g_ep=EgoyMDI0MTAyMS4xIKXMDSoASAFQAw%3D%3D). From my point of view gwl reach a recurring maximum during winter  and decline in summer due to evaporation from trees around, direclty "pumping" the shallow gw. The local geology is considered as poorly aquifer. I believe this looks like a wetland gw signature.

It is within a country side, without any pumping wells. This observation well is monitored and frequently visited by the French Geological Survey (BRGM).

I have tried various softwares and solutions (e.g. Pastas with FlexModel including an interception reservoir, and Gardenia with a double reservoir), however none of them allows to reproduce the summer declines. Winter fluctuations are quite easy to calculate. I am using ERA5-Land precipitations and PET. Maybe such PET under-estimates the potential evaporation from trees. 

Any suggestions would be welcome ! Regards

**Comment by mbakker7** [comment]
Thanks for the question @VeloVolant. Do you have a figure with the time series data and the fit produced with Pastas and Flexmodel? Maybe a results figure that also shows the fitted parameters? 
BTW, the link doesn't work (for me).

**Reply by dbrakenhoff** [reply]
I figured out how to see the graph yesterday. If you follow this link: https://ades.eaufrance.fr/Fiche/PtEau?Code=BSS001WHRD then click > "Mesures de niveau d'eau" -button at the top of the page and then the "Chronique piézométrique" button below that, you get to this graph: 

![image](https://github.com/user-attachments/assets/5c13c080-d7f9-4fe1-a4c6-20eb361140b4)

**Reply by VeloVolant** [reply]
You are a French master :)

**Reply by VeloVolant** [reply]
Here is the dataset
[all_data.xlsx](https://github.com/user-attachments/files/17519523/all_data.xlsx)

**Reply by VeloVolant** [reply]
import pandas as pd
import pastas as ps
import numpy as np
import matplotlib.pyplot as plt

oseries = pd.read_excel("all_data.xlsx", sheet_name="Niv", index_col=0, parse_dates=True).squeeze()
rain = pd.read_excel("all_data.xlsx", sheet_name="Era5_L_PP", index_col=0, parse_dates=True).squeeze()
evap = pd.read_excel("all_data.xlsx", sheet_name="Era5_L_PET", index_col=0, parse_dates=True).squeeze()
well =  pd.read_excel("all_data.xlsx", sheet_name="Era5_L_PET", index_col=0, parse_dates=True).squeeze()

ps.plots.series(oseries, [evap, rain])

ml = ps.Model(oseries.dropna().squeeze(), name="gwl")

rch = ps.rch.FlexModel(interception=True)
sm = ps.RechargeModel(rain, evap, recharge=rch, rfunc=ps.FourParam(), name="recharge")
ml.add_stressmodel(sm)

ml.solve(tmin="2000", tmax="2023")
ml.plots.results(figsize=(17, 9))
ml.plot(figsize=(17, 9), oseries=0, tmin="2000", tmax="2023")

**Reply by dbrakenhoff** [reply]
Hi @VeloVolant, sorry for never getting back to you after sharing your time series.

For what it's worth, I took a quick look at your data and model, and it's a tough one to simulate... 

When we see these time series with differing characteristics in dry/wet periods, we sometimes have success with the TarsoModel. See the notebook here: https://pastas.readthedocs.io/stable/examples/threshold_non_linear.html#tarsomodel. In that model there are two exponential responses, that are activated when the head reaches a certain level. So you can have a quick response in wet periods and a slow one in dry periods. 

But I tried that on your data and didn't get a good result. I think the reason is that your precipitation time series doesn't really look like it fits the observed head data. The grey line below is the rainfall (or rainfall - evaporation), the blue line represents pastas estimate of the contribution to head changes. The black dots are the measurements. The head seems to respond quickly to rainfall events, but these rainfall events are not really visible / timed correctly in the input stress. So that would make it extremely difficult to get a good model. So either the rainfall time series isn't representative for the location of your observation well, or something strange is going on there :)

<img width="1011" height="211" alt="image" src="https://github.com/user-attachments/assets/1d479e02-0c10-44d5-ba85-a7d88acfa6a7" />
